### PR TITLE
[DOC] Add docs for missing datasets

### LINKED ISSUE #3535: Missing Documentation for few datasets
## 📚 Documentation

I think we are missing documentation for a few datasets.

I verified over [master](https://pytorch.org/vision/master/datasets.html#) as well as over the [latest release](https://pytorch.org/vision/0.9/datasets.html), but couldn't find these three. I think they were considered as a stable feature in the release so we can add docs for them?

- [ ] [WiderFace](https://github.com/pytorch/vision/blob/master/torchvision/datasets/widerface.py) (included from latest release)
- [ ] [Caltech](https://github.com/pytorch/vision/blob/master/torchvision/datasets/caltech.py)
- [ ] [Semeion](https://github.com/pytorch/vision/blob/master/torchvision/datasets/semeion.py)

Do let me know! I can send a PR to fix all these three :smiley: 

cc @pmeier

### PR DESCRIPTION
**Pull request by oke-aditya**
Fixes #3535 .

I think there are few indentation errors in docs as well.
- [x] Add docs for Caltech
- [x] Add docs for WIDERFace
- [x] Add docs for SEMION 
- [x] Check Locally with Sphinx.

### CODE REVIEW

**Review by fmassa — COMMENTED** [review]
Thanks for the PR!

FYI doc building is currently failing for WiderFace, can you have a look?

**Comment by oke-aditya** [comment]
I fixed it like the above mentioned. It sadly shoes code :: Python in docs page. This happens even for Kinects case. Also I don't have any workaround currently.

File: torchvision/datasets/widerface.py [hunk 1]
```diff
@@ -14,6 +14,8 @@ class WIDERFace(VisionDataset):
     Args:
         root (string): Root directory where images and annotations are downloaded to.
             Expects the following folder structure if download=False:
+            .. code::
```
Comment (pmeier) [inline torchvision/datasets/widerface.py:17]: This should fix `.. code::` from appearing in the docs. 

```suggestion

            .. code::
```

Comment (oke-aditya) [reply torchvision/datasets/widerface.py:17]: Great. I will apply the same for Kinetics dataset too.

**Comment by oke-aditya** [comment]
I hope everything looks fine now. I tested it locally and it looked well. Let me know if I miss something!

**Review by pmeier — APPROVED** [review]
LGTM, thanks a lot  @oke-aditya!

**Pull request merged**